Add Github CI

endian-albin opened the pull request:
Use zephyr-west action

endian-albin commented:
@sdorre: What do you think?

endian-albin commented:
I suspect this part messes up caching:
`'init "-m https://github.com/endian-albin/pinetime-hypnos"'`

The original uses `'init "-l app"'`, but I could only make it work with the above.

petejohanson commented:
@endian-albin the key thing for the caching to work is to have the "root" of then *west* base be nested at least one directory from the root of your git repository.

If *not*, then the west stuff ends up putting the other dependencies outside of the workspace, and that messes everything up.

That's why ZMK puts the actual application under the `app` subdirectory.

The alternative was to `mv` a bunch of stuff in the next step after checkout and before the cache restore, to simulate that hierarchy even if it wasn't the hierarchy in the repo.

That could work as well, just complicates understanding the workflow logic and not confusing what is where after each step.

endian-albin commented:
@petejohanson: awesome, thanks. I'll try to reorganize the directories then.

sdorre commented:
Awesome ! 
Would it be possible to change the name of the release ? 
Something like `hypnos-fw` or whatever more explicit than artefact ? maybe with a commit hash to differenciate between build ?

It may be an improvement for later, but what about Making releases when creating a tag ?
Or even automatically tag the maser when a PR is merged ? Playing with the commit messages ? (Fix increment only minor version ? And feature increment major version ?)

endian-albin commented:
> Awesome ! 
> 
> Would it be possible to change the name of the release ? 
> 
> Something like `hypnos-fw` or whatever more explicit than artefact ?

Sure!

> maybe with a commit hash to differenciate between builds ?

Do you mean something like hypnos-fw-shortgithash? Sure, but I'd need do learn how to fetch that info first.

> 
> 
> 
> It may be an improvement for later, but what about Making releases when creating a tag ?

Yep.

> 
> Or even automatically tag the maser when a PR is merged ? Playing with the commit messages ? (Fix increment only minor version ? And feature increment major version ?) 

I think that such things would be difficult to automate, but lets start with things that feel natural. A tagged version such as 1.0 would not need the fit commit, but the commit just after this tag could have the short hash appended, or something like that.

Before all this, I'd like to reorganise the repo so as to facilitate caching though. Maybe I should wait doing that as long as we have other open PR's however?

sdorre commented:
> Before all this, I'd like to reorganise the repo so as to facilitate caching though. Maybe I should wait doing that as long as we have other open PR's however?

I don'd mind rebasing during work. If you are motivated. Then let's go. I'll have to wait forever otherwise.

endian-albin commented:
> > Before all this, I'd like to reorganise the repo so as to facilitate caching though. Maybe I should wait doing that as long as we have other open PR's however?
> 
> I don'd mind rebasing during work. If you are motivated. Then let's go. I'll have to wait forever otherwise.

OK then, I will try to move things, but I suspect that it will break the current "getting started" method in the README.  At least everything can be easily backed up.

petejohanson commented:
You can see how we handled our dev setup guide for ZMK where we have the subdirectory: https://zmkfirmware.dev/docs/dev-setup#initialize-west

endian-albin commented:
Progress (or not?) report:

- The artifacts archive is now named hypnos-fw-short-githash

- I moved the application to a subdirectory and tried to use the `west init -l app/` command in the Github Workflow. This wasn't compatible with the somewhat complex Kconfig overlay structure inherited from najnesnaj, however, and lead to the touch sensor driver being unrecognized. I then restored it to `west init -m git-repo-url` but kept the new directory structure and then it built. I don't know if this fixes caching or not; build time is almost the same every time anyway and not any different compared with the ZMK firmware builds (see their build history under Actions). I've not tried to build this locally since I actually don't wish to have zephyr and modules repos inside this repo. I'm not sure how to proceed.

endian-albin commented:
@sdorre, I fixed all the issues I believe and caching seems to be working. It takes about the same amount of time to build though. I've updated the "getting started" section in the README and did a few other adjustments there while at it. What do you think?

endian-albin commented:
Since we'll soon have P8 hardware support, I made some minor changes preparing for that.

sdorre commented on README.md:
```diff
@@ -15,12 +12,12 @@ and is inspired by [PineTime Hermes](https://github.com/Dejvino/pinetime-hermes-
 ## Features
 
 - [x] 100 % Free Software
-- [x] Low power consumption: 1.38 mA in idle mode (1.28 mA with the heart rate sensor disconnected), which gives a battery life of about one week.
```
Could be nice to get the exact measurement sometimes on the master, to see in which direction it goes. 
Maybe in the wiki ?

endian-albin replied:
Yes, I think the wiki is a good place for that. I no longer have a way to measure the consumption of my device. I could fix that later on though.

endian-albin commented:
> LGTM
> 
> Do we have a status on the DS D6 board ? I just saw it on the board folder and was wondering about hat.

Unknown. It was simply inherited from najnesnaj's project.
